pkg/config/yaml.go

```diff
@@ -1,19 +1,39 @@
 package config
 
 import (
+	"errors"
+	"fmt"
 	"io"
 	"os"
+	"regexp"
 	"strings"
 
 	"gopkg.in/yaml.v3"
 )
 
+var reYamlError = regexp.MustCompile(`line (\d+): field ([a-zA-Z0-9_-]+) not found .*$`)
+
 func LoadYAML(src string, obj interface{}) error {
 	configString := os.Expand(src, envReplace)
 	r := strings.NewReader(configString)
 	decoder := yaml.NewDecoder(r)
 	decoder.KnownFields(true)
 	if err := decoder.Decode(obj); err != nil {
+		// Test if the error is a yaml TypeError
+		if typeError, ok := err.(*yaml.TypeError); ok {
+			errs := []string{}
+			// Then extract the valuable data from the errors and make them more useful.
+			for _, invalidField := range typeError.Errors {
+				found := reYamlError.FindStringSubmatch(invalidField)
+				if len(found) == 3 {
+					line := found[1]
+					field := found[2]
+					errs = append(errs, fmt.Sprintf("line %s: field '%s' not expected", line, field))
+				}
+			}
+			fmt.Println(strings.Join(errs, ", "))
```

Comment:
Is this `fmt.Println` left over debugging code? Would use the returned error to contain the message we want and log it from the caller.

Reply:
Yes, thanks for catching it. There are tests to assert no errant output on successful invoke, but not for failures.